lib/bolt/transport/ssh/exec_connection.rb

```diff
@@ -12,8 +12,7 @@ def initialize(target)
           raise Bolt::ValidationError, "Target #{target.safe_name} does not have a host" unless target.host
 
           @target = target
-          ssh_config = Net::SSH::Config.for(target.host)
-          @user = @target.user || ssh_config[:user] || Etc.getlogin
+          @user = @target.user || Etc.getlogin
```

Comment:
I _think_ we want to omit `Etc.getlogin` here too and just rely on `ssh` doing the right thing. Otherwise this will _never_ respect the username set in SSH config and will always use the login user.

I'm not sure whether there's some reason I'm not thinking of why it's better to set the user to the login user explicitly...

Reply:
Ah this conversation has some context: https://github.com/puppetlabs/bolt/pull/1806#discussion_r423246069

So we _do_ need to know the SSH user for sure and can't just rely on `ssh` figuring it out.

Reply:
!!! I think we might be able to use `sshd -T` instead of `ssh -G` to do exactly that! It was introduced in OpenSSH 5.1, and I'm double checking our supported OS's all have at least that version but I'd be surprised if they didn't.

Reply:
It looks like that checks the sshd config file, which is different than the ssh config file.

Reply:
Sorry if that's a dumb question, but can't you just use $USER on the host ssh connects to for the things referenced in #1806? Let the destination shell evaluate it?

Reply:
@fuero There are some cases where that won't work, like some network devices and maybe powershell environments (which we have a feature for SSHing into, though it's experimental). @nicklewis had a great idea which was to try loading the config through Net::SSH and if it fails fall back to using `$USER`. Does that seem reasonable? I know it's a bit more work than what's here - let me know if you want to:
1. Make the changes yourself, 
2. Have me push the changes to your branch (I might need permission)
3. Have me put up a new PR with that behavior

Reply:
@fuero No rush, just wanted to check in and make sure this doesn't slip through the cracks. We'd like to get it fixed!

Reply:
@lucywyman Sorry, was busy at work. Here's another take, it works for my use case. Have a look!

Reply:
Perfect! One small syntax comment, then this looks good to me!